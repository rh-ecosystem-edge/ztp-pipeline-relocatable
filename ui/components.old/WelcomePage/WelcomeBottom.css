.welcome-bottom {
  align-items: center;
}

.welcome-bottom h1 {
  padding-top: 3rem;
}

.welcome-bottom:last-child {
  padding-bottom: 3rem;
}

.welcome-bottom__error {
  width: 40rem;
}
