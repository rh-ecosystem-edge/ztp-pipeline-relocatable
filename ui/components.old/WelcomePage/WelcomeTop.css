.welcome-top {
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
}

.welcome-top__kube {
  display: block;
  margin-left: auto;
  margin-right: auto;

  width: 188px; /* 18% */
  position: relative;
  top: -32px;
}
