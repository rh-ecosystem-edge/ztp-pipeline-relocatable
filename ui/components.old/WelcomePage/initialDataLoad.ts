import { getService } from '../../resources/service';
import { K8SStateContextData } from '../types';
import {
  SERVICE_TEMPLATE_API,
  SERVICE_TEMPLATE_METALLB_INGRESS,
} from '../PersistPage/resourceTemplates';
import { ipWithoutDots } from '../utils';
import { getHtpasswdIdentityProvider, getOAuth } from '../../resources/oauth';
import { workaroundUnmarshallObject } from '../../test-utils';
import { getIngressConfig } from '../../resources/ingress';
import { getClusterDomainFromComponentRoutes, Ingress } from '../../copy-backend-common';

export const initialDataLoad = async ({
  setNextPage,
  setError,
  handleSetApiaddr,
  handleSetIngressIp,
  handleSetDomain,
  setClean,
}: {
  setNextPage: (href: string) => void;
  setError: (message?: string) => void;
  handleSetApiaddr: K8SStateContextData['handleSetApiaddr'];
  handleSetIngressIp: K8SStateContextData['handleSetIngressIp'];
  handleSetDomain: K8SStateContextData['handleSetDomain'];
  setClean: K8SStateContextData['setClean'];
}) => {
  console.log('Initial data load');

  let ingressService, apiService, oauth;
  let ingressConfig: Ingress | undefined;

  try {
    oauth = await getOAuth().promise;
    ingressService = await getService({
      name: SERVICE_TEMPLATE_METALLB_INGRESS.metadata.name || '',
      namespace: SERVICE_TEMPLATE_METALLB_INGRESS.metadata.namespace || '',
    }).promise;
    apiService = await getService({
      name: SERVICE_TEMPLATE_API.metadata.name || '',
      namespace: SERVICE_TEMPLATE_API.metadata.namespace || '',
    }).promise;
    ingressConfig = await getIngressConfig().promise;
  } catch (_e) {
    const e = _e as { message: string; code: number };
    if (e.code !== 404) {
      console.error(e, e.code);
      setError('Failed to contact OpenShift Platform API.');
      return;
    }
  }

  // workarounds for tests
  oauth = workaroundUnmarshallObject(oauth);
  ingressService = workaroundUnmarshallObject(ingressService);
  apiService = workaroundUnmarshallObject(apiService);
  ingressConfig = workaroundUnmarshallObject(ingressConfig);

  handleSetIngressIp(
    ipWithoutDots(
      ingressService?.spec?.loadBalancerIP ||
        ingressService?.status?.loadBalancer?.ingress?.[0]?.ip,
    ),
  );
  handleSetApiaddr(
    ipWithoutDots(
      apiService?.spec?.loadBalancerIP || apiService?.status?.loadBalancer?.ingress?.[0]?.ip,
    ),
  );

  const currentHostname = getClusterDomainFromComponentRoutes(ingressConfig);
  if (currentHostname) {
    handleSetDomain(currentHostname);
  }

  setClean();

  if (getHtpasswdIdentityProvider(oauth)) {
    // The Edit flow for the 2nd and later run
    setNextPage('/settings');
  } else {
    // The Wizard for the very first run
    setNextPage('/wizard/username');
  }
};
