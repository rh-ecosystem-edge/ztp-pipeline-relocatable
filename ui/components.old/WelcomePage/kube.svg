<?xml version="1.0" encoding="UTF-8"?>
<svg width="238px" height="320px" viewBox="0 0 238 320" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <title>9431D01F-257E-4BE5-8828-C866B6042B3E</title>
    <g id="Concept-1---Approved" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Welcome" transform="translate(-841.000000, -188.000000)" fill-rule="nonzero">
            <g id="Modal" transform="translate(480.000000, 188.000000)">
                <g id="Illustration---Kube" transform="translate(361.000000, 0.000000)">
                    <polygon id="Path" fill="#DEE4E7" points="0 251.359631 118.113012 320 237.236439 251.382252 118.904755 182.711721"></polygon>
                    <g id="library-runtimes-grey" transform="translate(2.684387, 0.000000)">
                        <polygon id="Path" fill="#D1D1D1" opacity="0.5" points="0.467505537 67.8787879 0.233752769 203.040671 116.778359 135.991329 116.899006 4.08690325"></polygon>
                        <polygon id="Path" fill="#EBECEC" opacity="0.75" points="234.476648 201.18573 234.205193 67.7581413 116.899006 4.08690325 116.778359 135.991329"></polygon>
                        <polygon id="Path" fill="#DCDCDC" opacity="0.75" points="0.233752769 203.040671 116.778359 135.991329 234.552052 201.238513 116.899006 270.685706"></polygon>
                        <polygon id="Path" fill="#4C4C4C" points="227.071964 185.019087 117.735991 248.562138 7.26895707 184.90598 117.004571 121.476036"></polygon>
                        <polygon id="Path" fill="#B8B9B9" points="227.071964 185.019087 227.041802 197.211933 117.698289 260.754984 117.735991 248.562138"></polygon>
                        <polygon id="Path" fill="#DCDCDC" points="117.735991 248.562138 117.698289 260.754984 7.2237146 197.091286 7.26141665 184.90598"></polygon>
                        <polygon id="Path" fill="#EE0000" points="227.071964 167.110608 117.735991 230.653659 7.26895707 166.997502 117.004571 103.560017"></polygon>
                        <polygon id="Path" fill="#A30000" points="227.071964 167.110608 227.041802 179.295914 117.698289 242.838965 117.735991 230.653659"></polygon>
                        <polygon id="Path" fill="#CC0000" points="117.735991 230.653659 117.698289 242.838965 7.2237146 179.182808 7.26141665 166.989962"></polygon>
                        <polygon id="Path" fill="#DCDCDC" points="72.0260144 199.820915 165.036995 146.412178 165.036995 56.1157453 72.0260144 109.524483"></polygon>
                        <path d="M72.0260144,89.0447241 L72.0260144,108.182289 L165.029455,54.7660116 L165.029455,35.6284462 L72.0260144,89.0447241 Z M143.441255,61.0773363 C142.06136,61.8313775 140.937839,61.2281446 140.930298,59.6446581 C141.026586,57.8911237 141.953815,56.2887834 143.426175,55.3315425 C144.80607,54.5322588 145.929591,55.1731938 145.929591,56.7566803 C145.839783,58.5122402 144.914651,60.1186066 143.441255,61.0773363 L143.441255,61.0773363 Z M150.634808,56.9074886 C149.254913,57.7067722 148.131392,57.0658372 148.131392,55.4823507 C148.21869,53.7280621 149.144729,52.1229276 150.619728,51.1692351 C151.999623,50.3699515 153.123144,51.0108865 153.130685,52.594373 C153.030949,54.340847 152.103681,55.934851 150.634808,56.8848673 L150.634808,56.9074886 Z M157.866063,52.7527216 C156.486168,53.5520053 155.362647,52.9110703 155.355106,51.3275838 C155.447334,49.5727386 156.37188,47.9673919 157.843442,47.0069278 C159.230878,46.2151845 160.354399,46.8485791 160.354399,48.439606 C160.256366,50.1846774 159.332062,51.7784011 157.866063,52.7301004 L157.866063,52.7527216 Z" id="Shape" fill="#DCDCDC"></path>
                        <g id="container-grey" fill="#FFFFFF">
                            <path d="M3.67218059,69.4245723 L112.683915,132.130638 L113.347472,264.887129 L3.53645318,201.328998 L3.67218059,73.7980112 M0.143267826,67.7128988 L-1.67430778e-15,203.39507 L116.876384,271.032565 L116.167586,130.072105 L7.27649748,67.7355201 L0.143267826,67.7355201 L0.143267826,67.7128988 Z" id="Shape"></path>
                            <path d="M116.899006,4.08690325 L227.071964,67.7204392 L116.318394,130.15505 L7.2237146,67.6827372 L116.899006,4.08690325 M116.899006,0 L0.143267826,67.697818 L116.310853,134.219332 L234.205193,67.7581413 L116.899006,0 Z" id="Shape"></path>
                        </g>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>
