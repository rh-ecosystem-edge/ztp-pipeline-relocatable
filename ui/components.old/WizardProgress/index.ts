export * from './WizardProgress';
export * from './WizardProgressContext';
