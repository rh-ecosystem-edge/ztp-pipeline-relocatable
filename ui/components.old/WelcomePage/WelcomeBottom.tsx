import React from 'react';
import {
  Alert,
  AlertVariant,
  Button,
  ButtonVariant,
  Stack,
  StackItem,
  Title,
} from '@patternfly/react-core';
import { useNavigate } from 'react-router-dom';

import './WelcomeBottom.css';

export const WelcomeBottom: React.FC<{ error?: string; nextPage?: string }> = ({
  error,
  nextPage,
}) => {
  const navigate = useNavigate();

  return (
    <Stack className="welcome-bottom" hasGutter>
      <StackItem>
        <Title headingLevel="h1">Edge cluster</Title>
      </StackItem>
      <StackItem isFilled>
        To set up the configuration of your edge cluster, click Continue.
      </StackItem>
      {error && (
        <StackItem>
          <Alert
            title="Connection failed"
            variant={AlertVariant.danger}
            isInline
            className="welcome-bottom__error"
          >
            {error}
          </Alert>
        </StackItem>
      )}
      <StackItem>
        <Button
          component="a"
          data-testid="welcome-button-continue"
          onClick={() => navigate(nextPage || '/welcome')}
          variant={ButtonVariant.primary}
          isDisabled={!nextPage}
        >
          Continue
        </Button>
      </StackItem>
    </Stack>
  );
};
