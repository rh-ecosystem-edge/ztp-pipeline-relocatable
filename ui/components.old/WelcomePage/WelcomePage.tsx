import React from 'react';

import { Page } from '../Page';
import { ContentTwoRows } from '../ContentTwoRows';

import { WelcomeTop } from './WelcomeTop';
import { WelcomeBottom } from './WelcomeBottom';
import { useK8SStateContext } from '../K8SStateContext';
import { initialDataLoad } from './initialDataLoad';

export const WelcomePage: React.FC = () => {
  const [nextPage, setNextPage] = React.useState<string>();
  const [error, setError] = React.useState<string>();
  const { handleSetApiaddr, handleSetIngressIp, handleSetDomain, setClean } = useK8SStateContext();

  // Load initial data, switch between Inital vs. Edit flow
  React.useEffect(
    () => {
      initialDataLoad({
        setNextPage,
        setError,
        handleSetApiaddr,
        handleSetIngressIp,
        handleSetDomain,
        setClean,
      });
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [
      /* just once */
    ],
  );

  return (
    <Page>
      <ContentTwoRows
        top={<WelcomeTop />}
        bottom={<WelcomeBottom error={error} nextPage={nextPage} />}
      />
    </Page>
  );
};
