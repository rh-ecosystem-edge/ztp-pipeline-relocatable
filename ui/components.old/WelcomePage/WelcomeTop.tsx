import React from 'react';

import kube from './kube.svg';
import './WelcomeTop.css';

export const WelcomeTop: React.FC = () => (
  <div
    className="welcome-top"
    style={{
      backgroundImage: 'url("/backgroundWelcome.png")',
    }}
  >
    <img src={kube} className="welcome-top__kube" alt="Kube" />
  </div>
);
