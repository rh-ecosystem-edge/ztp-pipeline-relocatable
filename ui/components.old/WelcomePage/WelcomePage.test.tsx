import React from 'react';
import { render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { MemoryRouter } from 'react-router-dom';
import fetch from 'jest-fetch-mock';

import { WelcomePage } from './WelcomePage';
import { K8SStateContextProvider } from '../K8SStateContext';
import { WizardProgressContextProvider } from '../WizardProgress';
import { delay } from '../../test-utils';
import { IDENTITY_PROVIDER_NAME } from '../../copy-backend-common';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

const Component: React.FC = () => (
  <K8SStateContextProvider>
    <WizardProgressContextProvider>
      <MemoryRouter>
        <WelcomePage />
      </MemoryRouter>
    </WizardProgressContextProvider>
  </K8SStateContextProvider>
);

describe('Settings', () => {
  beforeEach(() => {
    fetch.resetMocks();
  });

  it('can render', async () => {
    let container;
    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(async () => {
      const { container: c } = render(<Component />);
      container = c;
      await delay(1000);
    });
    expect(fetch).toHaveBeenCalledTimes(4);

    expect(screen.getByTestId('welcome-button-continue')).not.toBeDisabled();
    expect(container).toMatchSnapshot();
  });

  it('can render wizard', async () => {
    fetch
      .once(JSON.stringify({ spec: {} })) /* OAuth */
      .once(JSON.stringify({ spec: { loadBalancerIP: undefined } }) /* ingress service */)
      .once(JSON.stringify({ spec: { loadBalancerIP: '123.1.2.3' } }) /* api service */)
      .once(JSON.stringify({ spec: { domain: 'apps.test.domain.com' } }) /* Ingress resource */);

    let container;
    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(async () => {
      const { container: c } = render(<Component />);
      container = c;
      await delay(1000);
    });
    expect(fetch).toHaveBeenCalledTimes(4);
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId('welcome-button-continue')).not.toBeDisabled();
    screen.getByTestId('welcome-button-continue').click();
    expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
    expect(mockedUsedNavigate.mock.calls[0][0]).toBe('/wizard/username');
  });

  it('can render Settings page', async () => {
    expect(mockedUsedNavigate).toHaveBeenCalledTimes(0);
    fetch
      .once(
        JSON.stringify({
          spec: {
            identityProviders: [
              {
                name: IDENTITY_PROVIDER_NAME,
                mappingMethod: 'claim',
                type: 'HTPasswd',
                htpasswd: {
                  fileData: {
                    name: 'a-file-name',
                  },
                },
              },
            ],
          },
        }),
      )
      .once(JSON.stringify({ spec: { loadBalancerIP: undefined } }) /* ingress service */)
      .once(JSON.stringify({ spec: { loadBalancerIP: '123.1.2.3' } }) /* api service */)
      .once(JSON.stringify({ spec: { domain: 'apps.test.domain.com' } }) /* Ingress resource */);

    let container;
    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(async () => {
      const { container: c } = render(<Component />);
      container = c;
      await delay(1000);
    });
    expect(fetch).toHaveBeenCalledTimes(4);
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId('welcome-button-continue')).not.toBeDisabled();
    screen.getByTestId('welcome-button-continue').click();
    expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
    expect(mockedUsedNavigate.mock.calls[0][0]).toBe('/settings');
  });
});
