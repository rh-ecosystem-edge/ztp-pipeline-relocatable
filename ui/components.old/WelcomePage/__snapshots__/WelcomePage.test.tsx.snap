// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings can render 1`] = `
<div>
  <div
    class="pf-c-background-image"
    style="--pf-c-background-image--BackgroundImage: url(/background.png); --pf-c-background-image--BackgroundImage-2x: url(/background.png); --pf-c-background-image--BackgroundImage--sm: url(/background.png); --pf-c-background-image--BackgroundImage--sm-2x: url(/background.png); --pf-c-background-image--BackgroundImage--lg: url(/background.png); --pf-c-background-image--Filter: url(#patternfly-background-image-filter-overlay0);"
  >
    <svg
      class="pf-c-background-image__filter"
      height="0"
      width="0"
      xmlns="http://www.w3.org/2000/svg"
    >
      <filter
        id="patternfly-background-image-filter-overlay0"
      >
        <fecolormatrix
          type="matrix"
          values="1 0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 1 0"
        />
        <fecomponenttransfer
          color-interpolation-filters="sRGB"
          result="duotone"
        >
          <fefuncr
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefuncg
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefuncb
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefunca
            tableValues="0 1"
            type="table"
          />
        </fecomponenttransfer>
      </filter>
    </svg>
  </div>
  <header
    class="pf-c-page__header page-header"
  >
    <div
      class="pf-c-page__header-brand"
    >
      <a
        class="pf-c-page__header-brand-link"
      >
        <img
          alt="Logo"
          src="RedHatLogo.svg"
        />
      </a>
    </div>
    <div
      class="pf-c-page__header-nav"
    >
      <div
        class="page-header-middle"
      >
        Edge cluster setup
      </div>
    </div>
    <div
      class="pf-c-page__header-tools"
    >
      <a
        aria-disabled="false"
        class="pf-c-button pf-m-tertiary"
        data-ouia-component-id="OUIA-Generated-Button-tertiary-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
      >
        Log out
      </a>
    </div>
  </header>
  <div
    class="page-container"
  >
    <div
      class="page-content"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <div
          class="pf-l-stack__item content-two-rows__top"
        >
          <div
            class="welcome-top"
            style="background-image: url(/backgroundWelcome.png);"
          >
            <img
              alt="Kube"
              class="welcome-top__kube"
              src="kube.svg"
            />
          </div>
        </div>
        <div
          class="pf-l-stack__item content-two-rows__bottom"
        >
          <div
            class="pf-l-stack pf-m-gutter welcome-bottom"
          >
            <div
              class="pf-l-stack__item"
            >
              <h1
                class="pf-c-title pf-m-2xl"
                data-ouia-component-id="OUIA-Generated-Title-1"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Edge cluster
              </h1>
            </div>
            <div
              class="pf-l-stack__item pf-m-fill"
            >
              To set up the configuration of your edge cluster, click Continue.
            </div>
            <div
              class="pf-l-stack__item"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-primary"
                data-ouia-component-id="OUIA-Generated-Button-primary-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="welcome-button-continue"
              >
                Continue
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings can render Settings page 1`] = `
<div>
  <div
    class="pf-c-background-image"
    style="--pf-c-background-image--BackgroundImage: url(/background.png); --pf-c-background-image--BackgroundImage-2x: url(/background.png); --pf-c-background-image--BackgroundImage--sm: url(/background.png); --pf-c-background-image--BackgroundImage--sm-2x: url(/background.png); --pf-c-background-image--BackgroundImage--lg: url(/background.png); --pf-c-background-image--Filter: url(#patternfly-background-image-filter-overlay2);"
  >
    <svg
      class="pf-c-background-image__filter"
      height="0"
      width="0"
      xmlns="http://www.w3.org/2000/svg"
    >
      <filter
        id="patternfly-background-image-filter-overlay2"
      >
        <fecolormatrix
          type="matrix"
          values="1 0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 1 0"
        />
        <fecomponenttransfer
          color-interpolation-filters="sRGB"
          result="duotone"
        >
          <fefuncr
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefuncg
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefuncb
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefunca
            tableValues="0 1"
            type="table"
          />
        </fecomponenttransfer>
      </filter>
    </svg>
  </div>
  <header
    class="pf-c-page__header page-header"
  >
    <div
      class="pf-c-page__header-brand"
    >
      <a
        class="pf-c-page__header-brand-link"
      >
        <img
          alt="Logo"
          src="RedHatLogo.svg"
        />
      </a>
    </div>
    <div
      class="pf-c-page__header-nav"
    >
      <div
        class="page-header-middle"
      >
        Edge cluster setup
      </div>
    </div>
    <div
      class="pf-c-page__header-tools"
    >
      <a
        aria-disabled="false"
        class="pf-c-button pf-m-tertiary"
        data-ouia-component-id="OUIA-Generated-Button-tertiary-3"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
      >
        Log out
      </a>
    </div>
  </header>
  <div
    class="page-container"
  >
    <div
      class="page-content"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <div
          class="pf-l-stack__item content-two-rows__top"
        >
          <div
            class="welcome-top"
            style="background-image: url(/backgroundWelcome.png);"
          >
            <img
              alt="Kube"
              class="welcome-top__kube"
              src="kube.svg"
            />
          </div>
        </div>
        <div
          class="pf-l-stack__item content-two-rows__bottom"
        >
          <div
            class="pf-l-stack pf-m-gutter welcome-bottom"
          >
            <div
              class="pf-l-stack__item"
            >
              <h1
                class="pf-c-title pf-m-2xl"
                data-ouia-component-id="OUIA-Generated-Title-3"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Edge cluster
              </h1>
            </div>
            <div
              class="pf-l-stack__item pf-m-fill"
            >
              To set up the configuration of your edge cluster, click Continue.
            </div>
            <div
              class="pf-l-stack__item"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-primary"
                data-ouia-component-id="OUIA-Generated-Button-primary-3"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="welcome-button-continue"
              >
                Continue
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings can render wizard 1`] = `
<div>
  <div
    class="pf-c-background-image"
    style="--pf-c-background-image--BackgroundImage: url(/background.png); --pf-c-background-image--BackgroundImage-2x: url(/background.png); --pf-c-background-image--BackgroundImage--sm: url(/background.png); --pf-c-background-image--BackgroundImage--sm-2x: url(/background.png); --pf-c-background-image--BackgroundImage--lg: url(/background.png); --pf-c-background-image--Filter: url(#patternfly-background-image-filter-overlay1);"
  >
    <svg
      class="pf-c-background-image__filter"
      height="0"
      width="0"
      xmlns="http://www.w3.org/2000/svg"
    >
      <filter
        id="patternfly-background-image-filter-overlay1"
      >
        <fecolormatrix
          type="matrix"
          values="1 0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 1 0"
        />
        <fecomponenttransfer
          color-interpolation-filters="sRGB"
          result="duotone"
        >
          <fefuncr
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefuncg
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefuncb
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefunca
            tableValues="0 1"
            type="table"
          />
        </fecomponenttransfer>
      </filter>
    </svg>
  </div>
  <header
    class="pf-c-page__header page-header"
  >
    <div
      class="pf-c-page__header-brand"
    >
      <a
        class="pf-c-page__header-brand-link"
      >
        <img
          alt="Logo"
          src="RedHatLogo.svg"
        />
      </a>
    </div>
    <div
      class="pf-c-page__header-nav"
    >
      <div
        class="page-header-middle"
      >
        Edge cluster setup
      </div>
    </div>
    <div
      class="pf-c-page__header-tools"
    >
      <a
        aria-disabled="false"
        class="pf-c-button pf-m-tertiary"
        data-ouia-component-id="OUIA-Generated-Button-tertiary-2"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
      >
        Log out
      </a>
    </div>
  </header>
  <div
    class="page-container"
  >
    <div
      class="page-content"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <div
          class="pf-l-stack__item content-two-rows__top"
        >
          <div
            class="welcome-top"
            style="background-image: url(/backgroundWelcome.png);"
          >
            <img
              alt="Kube"
              class="welcome-top__kube"
              src="kube.svg"
            />
          </div>
        </div>
        <div
          class="pf-l-stack__item content-two-rows__bottom"
        >
          <div
            class="pf-l-stack pf-m-gutter welcome-bottom"
          >
            <div
              class="pf-l-stack__item"
            >
              <h1
                class="pf-c-title pf-m-2xl"
                data-ouia-component-id="OUIA-Generated-Title-2"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Edge cluster
              </h1>
            </div>
            <div
              class="pf-l-stack__item pf-m-fill"
            >
              To set up the configuration of your edge cluster, click Continue.
            </div>
            <div
              class="pf-l-stack__item"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-primary"
                data-ouia-component-id="OUIA-Generated-Button-primary-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="welcome-button-continue"
              >
                Continue
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
